pkg/chartmuseum/server.go

```diff
@@ -154,11 +168,15 @@ func (server *Server) Listen(port int) {
 }
 
 func loggingMiddleware(logger *Logger) gin.HandlerFunc {
+	var requestID int64
 	return func(c *gin.Context) {
+		reqID := strconv.FormatInt(atomic.AddInt64(&requestID, 1), 10)
```

Comment:
What do these end up looking like?  like `"Request ID": 455` ? Will this result in collisions?

Reply:
Yes, this is the source of all requests, so the id's are unique. 1 to 2^64, I don't think there will be collisions (unless you get 100 req per second for 361 billion years :-)). Joking aside, no you won't get collisions because of the atomic increment for each request.

Reply:
Still, we probably want some sort of hash no? If the server is restarted, the first incoming request ID is "1", which will have been used for first request last time service started.

Also, if CM is run on multiple replicas, the "1" may be seen X times

Reply:
Yes, you are right, and I now see your point of view. I was not using the request-id for other purposes than in-process identification, so my joke was misplaced. How did you envision the utility of a globally unique request id (on the headers eventually) in chartmuseum ? From the debugging point of view, a uuid type of request-id would not be very practical in the current output in the logs (too long).

Reply:
I agree actually, which is why I suggested adding it as a k/v instead of log message prefix. The header would essentially be for tracking/debugging purposes, so you could match server-side logs to client requests.

Have you heard of jq? https://stedolan.github.io/jq/

If the logs are in JSON using `--log-json` option, you can pipe them to jq and format them however you want in the case that you are trying to debug request IDs visually

Reply:
Yes, jq is excellent. But I must admit that my itch was soothed once I got the incremental ids (for debugging concurrency). I think using jq would have slowed the iteration process for me at the moment. Would you mind if we opened an issue to implement the more global request-id as you want it ? I think it is not the core of what this PR is about.

Reply:
^ yes, good idea